lib/Parse/Lexer.cpp

```diff
@@ -1477,6 +1477,29 @@ void Lexer::lexImpl() {
       } else {
         diagnose(CurPtr-1, diag::lex_invalid_character)
           .fixItReplaceChars(getSourceLoc(CurPtr-1), getSourceLoc(tmp), " ");
+
+        StringRef confusedChar;
+        if (codepoint >= 0x80) {
+          llvm::SmallString<64> buffer;
+          EncodeToUTF8(codepoint, buffer);
+          confusedChar = buffer.str();
+        } else {
+          confusedChar = StringRef(std::string(1, (char)codepoint));
```

Comment:
Hmm...I'm a little confused as to what this means. Are you saying I should use `EncodeToUTF8()` regardless of whether the codepoint is greater than `0x80`? If so, that would mean changing the contractual semantics of the helper function, which worries me.